File: PN34.cpp


#include<iostream>
using namespace std;

class Number
{
    private:
        int iNo;       

    public:
        int iCnt = 0;
		
        void Accept()       
        {
            cout<<"Enter the value : "<<endl;
            cin>>this->iNo;
        }

        bool ChkPallindrome()
		{
			int iDigit = 0;
			int iRev = 0;
			int iTemp = 0;
			
			if(iNo < 0)
			{
				iNo = -iNo;
			}
			while(iNo > 0)
			{
				iDigit = iNo % 10;
				iRev = iRev * 10 + iDigit;
				iNo = iNo / 10;
			}
			if(iRev == iTemp)
			{
				return true;
			}
			else
			{
				return false;
			}
			
		}
};

int main()
{
    Number nobj;
	bool bRet;
	


    nobj.Accept();

	bRet = nobj.ChkPallindrome();
	
	if(bRet == true)
	{
		cout<<"It is pallindrome number<<endl: ";
		
	}
	else
	{
		cout<<"It is not pallindrome number"<<endl;
	}
	


    return 0;
}

File: PN34.java
import java.lang.*;
import java.util.*;

class Number
{
    private int iNo;

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter number : ");
        this.iNo = sobj.nextInt();
    }

    public void Display()
    {
        System.out.println("Value is : "+this.iNo);
    }

    public void FactRev()
    {
        int i = 0;
		
		for(i = iNo/2; i>= 1;i--)
		{
			if((iNo % i) == 0)
			{
				System.out.println(i);
			}
		}
			
           
    }
}

class PN34
{
    public static void main(String b[])
    {
            Number nobj = new Number();
            int iRet = 0;

            nobj.Accept();
            nobj.Display();

            nobj.FactRev();
            
    }
}